Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt

```diff
@@ -16,21 +16,15 @@
 
 package com.example.owl.ui
 
-import androidx.activity.OnBackPressedDispatcher
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.CompositionLocalProvider
-import com.example.owl.ui.utils.LocalBackDispatcher
 import com.example.owl.ui.utils.ProvideImageLoader
 import dev.chrisbanes.accompanist.insets.ProvideWindowInsets
 
 @Composable
-fun OwlApp(backDispatcher: OnBackPressedDispatcher) {
-
-    CompositionLocalProvider(LocalBackDispatcher provides backDispatcher) {
-        ProvideWindowInsets {
-            ProvideImageLoader {
-                NavGraph()
-            }
+fun OwlApp(finishActivity: () -> Unit) {
```

Comment:
Do we need to explicitly plumb this through? If we don't handle explicitly this will bubble up to the containing activity and finish it anyway?

Reply:
Not if the onboarding is being displayed, because the back stack has an additional destination in it at that point.